demisto_sdk/commands/pre_commit/hooks/hook.py

```diff
@@ -14,20 +19,21 @@ def __init__(
         self,
         hook: dict,
         repo: dict,
-        mode: str = "",
-        all_files: bool = False,
-        input_mode: bool = False,
+        context: "PreCommitContext",
     ) -> None:
         self.hooks: List[dict] = repo["hooks"]
         self.base_hook = deepcopy(hook)
         self.hook_index = self.hooks.index(self.base_hook)
         self.hooks.remove(self.base_hook)
-        self.mode = mode
-        self.all_files = all_files
-        self.input_mode = input_mode
+        self.mode = context.mode
+        self.all_files = context.all_files
+        self.input_mode = bool(context.input_files)
+        self.context = context
         self._set_properties()
+        self._exclude_hooks_by_version()
```

Comment:
would wrap these two in a function called `exclude_irrelevant_files` in case a subclass wants to override it easily

Reply:
done